lib/CMakeLists.txt

```diff
@@ -41,8 +41,16 @@ if (WIN32)
     )
 endif()
 
+set (LIBS
+    ${cpp-pcp-client_LIBRARY}
+    ${Boost_LIBRARIES}
+    ${OPENSSL_SSL_LIBRARY}
+    ${OPENSSL_CRYPTO_LIBRARY}
+    ${LEATHERMAN_LIBRARIES}
+)
+
 add_library(libpxp-agent STATIC ${LIBRARY_COMMON_SOURCES} ${LIBRARY_STANDARD_SOURCES})
-set_target_properties(libpxp-agent PROPERTIES PREFIX "" IMPORT_PREFIX "")
+target_link_libraries(libpxp-agent ${LIBS})
```

Comment:
Lack of `PRIVATE` flag means the libraries are associated with this target, and will be added to the link list for anything that links libpxp-agent. i.e. pxp-agent and unit tests.